components/bpmn/org.wso2.carbon.bpmn/src/main/java/org/wso2/carbon/bpmn/core/utils/SecretVaultValueReader.java

```diff
@@ -0,0 +1,149 @@
+            secretResolver = SecretResolverFactory.create(activitiConfigElement, true);
+
+        } catch (XMLStreamException | IOException e) {
+            log.error("Unable to read activiti.xml file", e);
+            throw new BPSFault("Failed to find activiti.xml", e);
+        }
+
+    }
+
+
+    /**
+     * Function that iterates through each property in the file and calls getSecretValue() to check if it has a
+     * secretAlias
+     *
+     * @return InputStream of the activiti.xml file with the resolved secretAliases. Returns null if there are no
+     * secret alias attributes in any of the properties
+     */
+    public InputStream decryptSecretValues() {
+
+        log.debug("Scanning activiti.xml for Secret Alias attribute");
+
+        boolean hasSecretValue = false;
+
+        //Gets all the beans in the Activiti.xml file
+        Iterator iteratorBeans = activitiConfigElement.getChildrenWithLocalName("bean");
+
+        //Iterates through all the beans in the iteratorBeans obj
+        while (iteratorBeans.hasNext()) {
+            Iterator iteratorProperties = ((OMElement) iteratorBeans.next()).getChildrenWithLocalName("property");
+
+            //Iterates through all the properties in the iteratorProperties obj
+            while (iteratorProperties.hasNext()) {
+                OMElement propertyOMElement = (OMElement) iteratorProperties.next();
+                String secretValue = getSecureVaultValue(secretResolver, propertyOMElement);
+
+                if (secretValue != null) {
+                    hasSecretValue = true;
+
+                    propertyOMElement.addAttribute("value", secretValue, null);
+                    propertyOMElement.removeAttribute(
+                            propertyOMElement.getAttribute(
+                                    new QName(BPMNConstants.SECURE_VAULT_NS, BPMNConstants.SECRET_ALIAS_ATTR_NAME)
+                            )
+                    );
+                }
+
+            } //End of iteration through iteratorProperties
+
+        } //End of iteration through iteratorBeans
+
+        log.debug("Scanning complete");
+
+        if (hasSecretValue) {
+            log.debug("Secret Alias found in activiti.xml");
+            InputStream activitiConfigStream = new ByteArrayInputStream(activitiConfigElement.toString().getBytes());
+            return activitiConfigStream;
+        } else {
+            log.debug("No Secret Alias found in activiti.xml");
+            return null;
+        }
+    }
+
+    /**
+     * Function to get the SecureVault value as a string.
+     *
+     * @param secretResolver  SecretResolver that resolves the password in the SecureVault
+     * @param propertyElement Parameter that needs to be resolved
+     * @return Resolved password as a String. Returns null if secretAlias attribute doesnt exist
+     */
+    public String getSecureVaultValue(SecretResolver secretResolver, OMElement propertyElement) {
```

Comment:
Can't we set access modifier to `private` here? Usage seems to within this class.

Reply:
Changed to private.